element-connector/src/main/java/org/eclipse/californium/elements/UDPConnector.java

```diff
@@ -282,32 +287,39 @@ public void run() {
 		 */
 		protected abstract void work() throws Exception;
 	}
-	
+
 	private class Receiver extends NetworkStageThread {
-		
+
 		private DatagramPacket datagram;
 		private int size;
-		
+
 		private Receiver(String name) {
 			super(name);
 			this.size = receiverPacketSize;
 			this.datagram = new DatagramPacket(new byte[size], size);
 		}
-		
+
 		protected void work() throws IOException {
 			datagram.setLength(size);
-			socket.receive(datagram);
-			LOGGER.debug("UDPConnector ({}) received {} bytes from {}:{}",
-					new Object[]{socket.getLocalSocketAddress(), datagram.getLength(),
-						datagram.getAddress(), datagram.getPort()});
-			byte[] bytes = Arrays.copyOfRange(datagram.getData(), datagram.getOffset(), datagram.getLength());
-			RawData msg = RawData.inbound(bytes, new AddressEndpointContext(datagram.getAddress(), datagram.getPort()), false);
-
-			receiver.receiveData(msg);
+			DatagramSocket socket;
+			synchronized (this) {
```

Comment:
What is the purpose of those synchronized block

Reply:
to ensure, that the synchronized start/stop has finished.

Reply:
:man_facepalming:  I missed that start and stop was synchronised method ...

Reply:
I got in rare case NullPointerExceptiosn, therefore a touched the socket handling.